src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.cs

```diff
@@ -77,5 +79,68 @@ public async Task ItemsSourceDoesNotLeak()
 			Assert.NotNull(logicalChildren);
 			Assert.True(logicalChildren.Count <= 3, "_logicalChildren should not grow in size!");
 		}
+
+		[Fact]
+		public async Task ClearingItemsSourceClearsBindingContext()
+		{
+			SetupBuilder();
+
+			IReadOnlyList<Element> logicalChildren = null;
+			var collectionView = new CollectionView
+			{
+				ItemTemplate = new DataTemplate(() => new Label() {  HeightRequest = 30, WidthRequest = 200 }),
+				WidthRequest = 200,
+				HeightRequest = 200,
+			};
+
+			await CreateHandlerAndAddToWindow<CollectionViewHandler>(collectionView, async handler =>
+			{
+				var data = new ObservableCollection<MyRecord>()
+				{
+					new MyRecord("Item 1"),
+					new MyRecord("Item 2"),
+					new MyRecord("Item 3"),
+				};
+				collectionView.ItemsSource = data;
+				await Task.Delay(100);
+
+				logicalChildren = collectionView.LogicalChildrenInternal;
+				Assert.NotNull(logicalChildren);
+				Assert.True(logicalChildren.Count == 3);
+
+				// Clear collection
+				var savedItems = data.ToArray();
+				data.Clear();
+
+				await Task.Delay(100);
```

Comment:
No GC's need to occur on these `Task.Delay()` calls? Is it just giving a chance for something to run on the UI thread?

I'm just wondering how we know 100ms is enough, or will it be a flaky test?

Reply:
Just needs to run layout on UI thread. Likely `Task.Yield()` would have been enough but I followed what the other tests used.

Reply:
The other tests are probably also wrong, lol. Maybe we should make a helper method for something like:
```csharp
await WaitForMainThread();
// Which does something like this underneath?
// await InvokeOnMainThreadAsync(() => { });
```

Reply:
Incidentally, that's what `Task.Yield()` does (as long as we are running in the correct synchronization context, which is the case here).

Reply:
@PureWeen  did  we add a helper for this?